File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = prices[0]
        max_profit = 0

        for i in prices[1:]:
            if i < min_price:
                min_price = i
            else:
                profit = i - min_price
                if profit > max_profit:
                    max_profit = profit
        return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
	minPrice := prices[0]
	maxProfit_ := 0

	for i := range prices {
		if prices[i] < minPrice {
			minPrice = prices[i]
		} else if prices[i]-minPrice > maxProfit_ {
			maxProfit_ = prices[i] - minPrice
		}
	}
	return maxProfit_
}